### DISCUSSION TITLE
Movement to edit commands

### DISCUSSION THREAD

**Discussion by Tylopilus** [wezterm/wezterm#6393] [Q&A]
### What Operating System(s) are you running on?

Windows

### WezTerm version

20240203-110809-5046fc22 x86_64-pc-windows-msv

### Ask your question!

How can I efficently navigate around in commands to edit them? E.g. I run a command and the want to rerun it with a slightly different config.

so I hit uparrow to bring the old command back but then I cannot use
ctrl + leftarrow to jump over words
cannot click with my mouse to place the carrot
cannot remove whole words by pressing ctrl+backspace

So how do you actually use wezterm for this? Is there a VI-Mode that I can jump into to move around and delete/edit words/cursor positions?

**Comment by tcchau** [comment]
I'm not sure why no one has answered this, and I'm not sure if it will work in the Windows version, as I'm on a Mac, but I get into VI-mode by hitting ESC after which I can use the usual edit mode VI keybindings to edit the command line. The problem I run into is that every time I hit ESC, the terminal deletes an existing line on the screen.

**Reply by tcchau** [reply]
Thanks for that. I'm not the OP so it doesn't really apply to me... I am 90% on MacOS, 6% on Windows and 4% on Ubuntu. I've always used "vi mode" ilne editing so I didn't even know the Ctrl-arrow keys were a common binding for most people.

But I did figure out my disappearing terminal lines problem. It had nothing to do with Wezterm. 

I finally took some time to troubleshoot it tonight after having lived with it for months. It turns out that my powerlevel10k theme was clashing with some zle bindings and commands I had put into my .zshrc way back before I started using oh-my-zsh and powerlevel10k, making the zle stuff obsolete. I removed those, and now I'm feeling more productive!
